File: app/src/main/java/com/arthur/gloria/logired/features/trip/mytrips/data/repository/MyTripsRepositoryImpl.kt
package com.arthur.gloria.logired.features.trip.mytrips.data.repository

import com.arthur.gloria.logired.core.network.ApiService
import com.arthur.gloria.logired.core.network.model.Trip
import com.arthur.gloria.logired.features.trip.mytrips.domain.repository.MyTripsRepository
import javax.inject.Inject

class MyTripsRepositoryImpl @Inject constructor(
    private val api: ApiService
) : MyTripsRepository {

    override suspend fun getMyTrips(): Result<List<Trip>> {
        return try {
            val response = api.getMyRequestedTrips()
            if (response.isSuccessful && response.body() != null) {
                Result.success(response.body()!!.rides ?: emptyList())
            } else {
                Result.failure(Exception("Error al obtener viajes: ${response.code()}"))
            }
        } catch (e: Exception) {
            Result.failure(Exception("Error de conexión: ${e.message}"))
        }
    }
}

File: lib/features/trip/mytrips/data/repository/my_trips_repository_impl.dart
import '../../../../../core/network/api_service.dart';
import '../../../../../core/network/model/models.dart';
import '../../../../../core/utils/ride_status.dart';
import '../../domain/my_trips_repository.dart';

class MyTripsRepositoryImpl implements MyTripsRepository {
  final ApiService _api;

  MyTripsRepositoryImpl(this._api);

  @override
  Future<List<Trip>> getMyTrips() async {
    final response = await _api.getMyRequestedTrips();
    final list = response.data['rides'] as List? ?? [];
    return list.map((e) => Trip.fromJson(e)).toList();
  }

  @override
  Future<void> cancelTrip(int tripId, {int reason = CancelReason.manual}) async {
    await _api.updateRideStatus(tripId, {
      'status': RideStatus.cancelled,
      'cancel_reason': reason,
    });
  }
}
